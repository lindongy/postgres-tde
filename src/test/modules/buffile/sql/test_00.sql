CREATE EXTENSION IF NOT EXISTS buffile;

-- This test only verifies that PG is compiled with a component file size of
-- 32 kB (i.e. 4 buffers of 8 kB) instead of 1 GB. That seems appropriate for
-- testing. Some other tests may rely on it.
BEGIN;
SELECT buffile_create();
-- Skip the first component file.
SELECT buffile_seek(0, 32768);
-- Write the first byte of the second component file. We can't simply
-- buffile_seek() beyond the position 32768 as it would return EOF.
SELECT buffile_write('a');
-- Enforce BufFileFlush(), which actually adds the component file.
SELECT buffile_read(1);
-- Check that we're in the 2nd file, i.e. the file size is as expected.
SELECT buffile_assert_fileno(1);
SELECT buffile_close();
COMMIT;
