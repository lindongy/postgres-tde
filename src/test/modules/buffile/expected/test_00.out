CREATE EXTENSION IF NOT EXISTS buffile;
-- This test only verifies that PG is compiled with a component file size of
-- 32 kB (i.e. 4 buffers of 8 kB) instead of 1 GB. That seems appropriate for
-- testing. Some other tests may rely on it.
BEGIN;
SELECT buffile_create();
 buffile_create 
----------------
 
(1 row)

-- Skip the first component file.
SELECT buffile_seek(0, 32768);
 buffile_seek 
--------------
            0
(1 row)

-- Write the first byte of the second component file. We can't simply
-- buffile_seek() beyond the position 32768 as it would return EOF.
SELECT buffile_write('a');
 buffile_write 
---------------
             1
(1 row)

-- Enforce BufFileFlush(), which actually adds the component file.
SELECT buffile_read(1);
 buffile_read 
--------------
 \x
(1 row)

-- Check that we're in the 2nd file, i.e. the file size is as expected.
SELECT buffile_assert_fileno(1);
 buffile_assert_fileno 
-----------------------
 
(1 row)

SELECT buffile_close();
 buffile_close 
---------------
 
(1 row)

COMMIT;
